server/src/test/java/org/elasticsearch/http/AbstractHttpServerTransportTests.java

```diff
@@ -1118,8 +1117,16 @@ public void testForceClosesOpenChannels() {
                 } catch (InterruptedException e) {
                     fail("server never called grace period");
                 }
-                // one last request, will attempt to close naturally, but we are blocking it
-                transport.incomingRequest(new TestHttpRequest(HttpRequest.HttpVersion.HTTP_1_1, RestRequest.Method.GET, "/"), httpChannel);
+                try {
+                    // one last request, will attempt to close naturally, but we are blocking it
+                    transport.incomingRequest(
+                        new TestHttpRequest(HttpRequest.HttpVersion.HTTP_1_1, RestRequest.Method.GET, "/"),
+                        httpChannel
+                    );
+                } catch (IllegalStateException err) {
+                    // we force closed below, so we may get this error when closing the channel after request succeeds.
```

Comment:
This was changed to an `AssertionError` by #96564

Reply:
> Seems like we should not double close? That's why we have the assertion, to ensure we aren't double closing. :)

If we force close from outside the channel, the channel is going to also attempt to close.  That's the nature of it being forced.

Reply:
Perhaps it is better to consistently close from our end then, rather than (sometimes) sending `Connection: close`?

Reply:
Setting `Connection: close` is the graceful way to close http connections.  We want to give these requests a chance to close naturally.  Force closing everything is the way we did this before the graceful shutdown feature.

Reply:
Graceful shutdown to me is about allowing already running requests to complete. Asking the other end of the http channel to close the connection rather than us closing from our end seems tangential.

Reply:
>  Asking the other end of the http channel to close the connection rather than us closing from our end seems tangential.

Closing the connection occurs on our side with graceful shutdown, we send the headers to inform the client that we are going to close it.

Reply:
Sure, but we don't do this in all cases, it is best effort right?

That aside, if we are closing on our side, then we should be able to avoid double closing? We added the double close assertion so that we ensure we aren't unexpectedly double closing (logic error). Catching an IllegalStateException (which must be wrapping the assertion?), which shouldn't occur in production, seems to have the possibility to hide real problems.

Reply:
For this test, we are no longer catching the `IllegalStateException`.  Instead, when we block the close in the test channel and timeout, we simply return.